sympy/core/power.py

```diff
@@ -582,6 +582,9 @@ def _check(ct1, ct2, old):
                     combines = Pow._eval_power(
                         Pow(*old.as_base_exp(), evaluate=False),
                         pow) is not None
+                if not combines:
+                    if pow.has(pi):
```

Comment:
the variable `combines = False` doesnt allow for the exponential expression to be substituted in the `subs()` method. I am wondering the use of [this](https://github.com/sympy/sympy/pull/11225/files#diff-1929ff6261eb677c1870249eafa0f6e5R578) method. Why does the ratio of exponent has to be an integer? Since we are not looking for the exact expression to be replaced, why cant the substituted expression have non-integral exponents?